CardScan/Classes/CreditCardOcr/OcrMainLoop.swift

```diff
@@ -92,21 +92,30 @@ public class OcrMainLoop : MachineLearningLoop {
     var userDidCancel = false
     
     public init(analyzers: [AnalyzerType] = [.legacy, .apple]) {
-        scanStats.model = "legacy+apple"
-        machineLearningQueues = []
+        var ocrImplementations: [CreditCardOcrImplementation] = []
         for analyzer in analyzers {
             let queue = DispatchQueue(label: "\(analyzer) OCR ML")
             switch (analyzer) {
             case .legacy:
                 if #available(iOS 11.2, *) {
-                    analyzerQueue.append(LegacyCreditCardOcr(dispatchQueue: queue))
+                    ocrImplementations.append(LegacyCreditCardOcr(dispatchQueue: queue))
                 }
             case .apple:
                 if #available(iOS 13.0, *) {
-                    analyzerQueue.append(AppleCreditCardOcr(dispatchQueue: queue))
+                    ocrImplementations.append(AppleCreditCardOcr(dispatchQueue: queue))
                 }
             }
-            machineLearningQueues.append(queue)
+        }
+        setupMl(ocrImplementations: ocrImplementations)
+    }
+    
+    /// Note: you must call this function in your constructor
+    public func setupMl(ocrImplementations: [CreditCardOcrImplementation]) {
+        machineLearningQueues = []
+        scanStats.model = "legacy+apple"
```

Comment:
should the scan stats model be called legacy+apple if some devices can support legacy but not apple?

Reply:
the system is still "legacy+apple" but not all devices get apple